string[] objects
---
float64[] probabilities
